Description: fix missing icon for launchers that have a period (dot, '.') in their icon's base name
Author: Gauvain Pocentek <gauvain@ulteo.com>
Bug: https://bugzilla.xfce.org/show_bug.cgi?id=3614

Index: libxfcegui4-4.8.1/libxfcegui4/icons.c
===================================================================
--- libxfcegui4-4.8.1.orig/libxfcegui4/icons.c	2011-02-10 10:33:27.000000000 +0100
+++ libxfcegui4-4.8.1/libxfcegui4/icons.c	2011-09-05 21:05:46.094685532 +0200
@@ -142,8 +142,12 @@
     ensure_gtk_icon_theme();
 
     /* GtkIconTheme doesn't like extensions */
-    if((p = g_strrstr(name, ".")) && strlen(p) < 6)
-        name_fixed = g_strndup(name, p-name);
+    if (g_str_has_suffix (name, ".png") || g_str_has_suffix (name, ".svg")
+        || g_str_has_suffix (name, ".xpm"))
+    {
+        if((p = g_strrstr(name, ".")) && strlen(p) < 6)
+            name_fixed = g_strndup(name, p-name);
+    }
     
     pix = gtk_icon_theme_load_icon(icon_theme, name_fixed ? name_fixed : name,
                                    size,
@@ -220,9 +224,13 @@
     ensure_gtk_icon_theme();
     
     /* GtkIconTheme doesn't like extensions */
-    if((p = g_strrstr(name, ".")) && strlen(p) < 6)
-        name_fixed = g_strndup(name, p-name);
-    
+    if (g_str_has_suffix (name, ".png") || g_str_has_suffix (name, ".svg")
+        || g_str_has_suffix (name, ".xpm"))
+    {   
+        if((p = g_strrstr(name, ".")) && strlen(p) < 6)
+            name_fixed = g_strndup(name, p-name);
+    }   
+ 
     info = gtk_icon_theme_lookup_icon(icon_theme,
                                       name_fixed ? name_fixed : name,
                                       size, 0);
